observelib/observe/src/main/ballerina/modules/mockextension/metrics.bal

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/log;
+
+public type Tag record {
+  string key;
+  string value;
+  int hashCode;
+};
+
+public type MetricId record {
+  string name;
+  string description;
+  Tag[] tags;
```

Comment:
Shall we make this a map of strings ?

Reply:
Tag contains an int field `hashcode`, therefore its not possible to make it a map

Reply:
We can drop hashcode and use it. `hashcode` is used to increase the performance of set / map operations mainly in the runtime.
It keeps a pre-calculated hash value without re-creating the hash each and every time a tag is put into a set / map

Reply:
As per the offline discussion, dropped `hashcode` fields from records and kept the Tag record as it is not a map.